### ISSUE TITLE
Incorrect typing for tables sent through RemoteEvents

### ISSUE THREAD

**Issue by HJWoods** [roblox-ts/roblox-ts#2286]
I have a feeling that when passed through remote events, the meta-table of a table is discarded, probably to stop remote code injection or something like that.

I have a class "GameObject" which is sent from the player to the server and vice versa. If you try and invoke a method of a class that has been sent through a remote event, like in this example:
![image](https://user-images.githubusercontent.com/38426334/218206085-d3e203f1-cc53-4420-a241-41834547cd8e.png)
![image](https://user-images.githubusercontent.com/38426334/218206316-9e685e25-8564-4945-8df2-5eb7004bb138.png)
The following happens:
![image](https://user-images.githubusercontent.com/38426334/218206341-0676744b-9b5e-4622-9146-336b1730de8f.png)
Invoking properties is fine, it's only if you try and call a function, which is why I suspect the name is likely being discarded.
I'm aware that passing functions around like this is a bad idea anyway, but this confused me for a fair while, and I would argue that the typing for classes sent thru a remote event in this way should reflect the fact that you are only able to invoke non-function properties of the table.
In the unlikely event someone else has the same problem, the best workaround I could come up with is to make a static method for the class, like this
![image](https://user-images.githubusercontent.com/38426334/218206890-1108ec94-a0e3-4b51-a4d1-8269865d60a1.png)

**Comment by safazi** [comment]
You cannot send functions over a remote event or function, and yes, the metatable is included in that.

**Comment by HJWoods** [comment]
Right, but the typing currently doesn't reflect that, and it's not immediately clear what the problem is unless you are familiar with how the underlying lua works, which I would say is less than ideal
